api/prometheus/v1/api.go

```diff
@@ -818,10 +818,33 @@ func (h *httpAPI) LabelValues(ctx context.Context, label string, matches []strin
 	return labelValues, w, json.Unmarshal(body, &labelValues)
 }
 
-func (h *httpAPI) Query(ctx context.Context, query string, ts time.Time) (model.Value, Warnings, error) {
+type apiOptions struct {
+	timeout time.Duration
+}
+
+type Option func(c *apiOptions)
+
+func WithTimeout(timeout time.Duration) Option {
+	return func(o *apiOptions) {
+		o.timeout = timeout
+	}
+}
+
+func (h *httpAPI) Query(ctx context.Context, query string, ts time.Time, opts ...Option) (model.Value, Warnings, error) {
+
 	u := h.client.URL(epQuery, nil)
 	q := u.Query()
 
+	opt := &apiOptions{}
+	for _, o := range opts {
+		o(opt)
+	}
+
+	t := opt.timeout.Milliseconds()
+	if t > 0 {
+		q.Set("timeout", fmt.Sprint(t))
```

Comment:
```suggestion
		q.Set("timeout", t.String())
```
I think we can just pass the duration, and let the server handle it https://github.com/prometheus/prometheus/blob/0b41fd6e716f3e8b79016f73b6dd77fdf5fcdced/web/api/v1/api.go#L1598-L1610

https://github.com/prometheus/common/blob/0c7319ad920e3239143af6c9f87149ff05faf2ab/model/time.go#L188-L234